api/src/DataPersister/PeriodDataPersister.php

```diff
@@ -19,9 +22,56 @@ public function __construct(
         );
     }
 
+    public function beforeCreate($data): BaseEntity {
+        static::managePeriodDays($data);
+
+        return $data;
+    }
+
+    public function beforeUpdate($data): BaseEntity {
+        $orig = $this->em->getUnitOfWork()->getOriginalEntityData($data);
+
+        static::managePeriodDays($data, $orig);
+
+        return $data;
+    }
+
     public function beforeRemove($data): ?BaseEntity {
         $this->validator->validate($data, ['groups' => ['delete', 'Period:delete']]);
 
         return null;
     }
+
+    public static function managePeriodDays(Period $period, array $orig = null) {
+        $length = $period->getPeriodLength();
+        $days = $period->getDays();
+
+        $minDelta = 0;
+        if (null != $orig) {
+            $minDelta = $orig['start']->getTimestamp() - $period->start->getTimestamp();
+            $minDelta = floor($minDelta / 60);
+        }
+
+        // Add Days
+        $i = count($days);
```

Comment:
Wäre es schlimm wenn wir mal aus Versehen in der DB inkonsistente Daten haben und `count($days)` nicht gleich ist wie der grösste `dayOffset` aller Days dieser Periode?
Ich frage mich, ob es hier angebracht wäre, so einen Zustand zu "reparieren", oder ob das besser anderswo gemacht würde, wenn überhaupt.